src/getCartReceptors.C

```diff
@@ -127,21 +135,81 @@ void MeshBlock::getCartReceptors(CartGrid *cg,parallelComm *pc)
 		      pmap[cg->proc_id[c]]=1;
 		      dataPtr->next=(INTEGERLIST2 *) malloc(sizeof(INTEGERLIST2));
 		      dataPtr=dataPtr->next;
-		      dataPtr->realData=(double *) malloc(sizeof(double)*3);
-		      dataPtr->intData=(int *) malloc(sizeof(int)*3);
+          dataPtr->intDataSize=4;
+          dataPtr->realDataSize=4;
+          dataPtr->realData=(double *) malloc(sizeof(double)*dataPtr->realDataSize);
+          dataPtr->intData=(int *) malloc(sizeof(int)*dataPtr->intDataSize);
 		      dataPtr->intData[0]=cg->proc_id[c];
 		      dataPtr->intData[1]=cg->local_id[c];
 	        dataPtr->intData[2]=itm;
-		      dataPtr->intDataSize=3;
-		      dataPtr->realDataSize=1;
+	        dataPtr->intData[3]=0;
 		      nsearch+=1;
 
-			  for(kk=0;kk<3;kk++)
-			    dataPtr->realData[kk]=xtm[kk];
+          for(kk=0;kk<dataPtr->realDataSize-1;kk++)
+            dataPtr->realData[kk]=xtm[kk];
+
+          dataPtr->realData[dataPtr->realDataSize-1]=vol;
 
 		      dataPtr->next=NULL;
 		    }
 		}
+
+    // loop over all nodes
+    for(j=0;j<cg->dims[3*c]+1;j++)
+      for(k=0;k<cg->dims[3*c+1]+1;k++)
+        for(l=0;l<cg->dims[3*c+2]+1;l++)
+        {
+          // Q[nq,nZ+1+2*nf,nY+1+2*nf,nX+1+2*nf]--> C++ Node storage
+          // indexing starts from number of cells onwards
+          itm = (cg->dims[3*c+1]+1+2*cg->nf)*(cg->dims[3*c]+1+2*cg->nf)*(l+cg->nf)
+              + (cg->dims[3*c]+1+2*cg->nf)*(k+cg->nf) + (j+cg->nf)
+              + cell_count;
```

Comment:
I recommend that we create two `inline` functions `get_cell_index` and `get_node_index` that computes this. I understand that it feels like we are bringing back `get_amr_index_xyz` but that function did too many things.

Reply:
@sayerhs I have introduced the functions throughout the code in d1bb7f7